tests/deprecation/test_deprecate_posargs.py

```diff
@@ -0,0 +1,464 @@
+import inspect
+
+from django.test import SimpleTestCase
+from django.utils.deprecation import RemovedAfterNextVersionWarning, deprecate_posargs
+
+
+class DeprecatePosargsTests(SimpleTestCase):
```

Comment:
Reviewer note: please leave feedback on these tests for `@deprecate_posargs` in #19145, not this PR.